File: freecodecamp/daily-code-challenge/2026-02-21.py
def score_curling(house):
    center = (2, 2)
    
    def ring_distance(r, c):
        return max(abs(r - center[0]), abs(c - center[1]))
    
    stones = {"R": [], "Y": []}
    
    for r in range(5):
        for c in range(5):
            if house[r][c] in stones:
                stones[house[r][c]].append(ring_distance(r, c))
    
    if not stones["R"] and not stones["Y"]:
        return "No points awarded"
    
    if not stones["R"]:
        scoring_team = "Y"
    elif not stones["Y"]:
        scoring_team = "R"
    else:
        min_r = min(stones["R"])
        min_y = min(stones["Y"])
        
        if min_r == min_y:
            return "No points awarded"
        
        scoring_team = "R" if min_r < min_y else "Y"
    
    opponent = "Y" if scoring_team == "R" else "R"
    opponent_min = min(stones[opponent]) if stones[opponent] else float("inf")
    
    points = sum(1 for d in stones[scoring_team] if d < opponent_min)
    
    return f"{scoring_team}: {points}"

File: freecodecamp/daily-code-challenge/2026-02-21.js
function scoreCurling(house) {
  const center = [2, 2];
  
  function ringDistance(r, c) {
    return Math.max(Math.abs(r - center[0]), Math.abs(c - center[1]));
  }
  
  const stones = { R: [], Y: [] };
  
  for (let r = 0; r < 5; r++) {
    for (let c = 0; c < 5; c++) {
      if (house[r][c] === "R" || house[r][c] === "Y") {
        stones[house[r][c]].push(ringDistance(r, c));
      }
    }
  }
  
  if (stones.R.length === 0 && stones.Y.length === 0) {
    return "No points awarded";
  }
  
  let scoringTeam;
  
  if (stones.R.length === 0) {
    scoringTeam = "Y";
  } else if (stones.Y.length === 0) {
    scoringTeam = "R";
  } else {
    const minR = Math.min(...stones.R);
    const minY = Math.min(...stones.Y);
    
    if (minR === minY) {
      return "No points awarded";
    }
    
    scoringTeam = minR < minY ? "R" : "Y";
  }
  
  const opponent = scoringTeam === "R" ? "Y" : "R";
  const opponentMin = stones[opponent].length > 0
    ? Math.min(...stones[opponent])
    : Infinity;
  
  const points = stones[scoringTeam].filter(d => d < opponentMin).length;
  
  return `${scoringTeam}: ${points}`;
}
